builder/simplify_test.cc

```diff
@@ -130,16 +130,40 @@ TEST(simplify, basic) {
 
 TEST(simplify, let) {
   // lets that should be removed
-  test_simplify(let::make(0, y, z), z);                      // Dead let
-  test_simplify(let::make(0, y * 2, x), y * 2);              // Single use, substitute
-  test_simplify(let::make(0, y, (x + 1) / x), (y + 1) / y);  // Trivial value, substitute
-  test_simplify(let::make(0, 10, x / x), 1);                 // Trivial value, substitute
+  test_simplify(let::make(0, y, z), z);                                // Dead let
+  test_simplify(let::make(0, y * 2, x), y * 2);                        // Single use, substitute
+  test_simplify(let::make(0, y, (x + 1) / x), (y + 1) / y);            // Trivial value, substitute
+  test_simplify(let::make(0, 10, x / x), 1);                           // Trivial value, substitute
+  test_simplify(let::make(0, buffer_max(y, 0), x), buffer_max(y, 0));  // buffer_max, substitute
+  test_simplify(let::make(0, buffer_min(y, 0), x), buffer_min(y, 0));  // buffer_min, substitute
 
   // lets that should be kept
   test_simplify(
       let::make(0, y * 2, (x + 1) / x), let::make(0, y * 2, (x + 1) / x));  // Non-trivial, used more than once.
 }
 
+TEST(simplify, let_in_loop) {
+  expr let = let::make(0, y, z);
```

Comment:
These tests aren't really testing the new code, because the let is trivial in every case. I think we need a few tests, one for each of these cases:
- `let y = z in loop(x, [0, 3], check::make(y)` --> substituted let (trivial)
- `let y = z*w in loop(x, [0, 3], check::make(y)` --> not substituted let (used in loop)
- `let y = z*w in check(y)` --> substituted let (one use)
- `let y = z*w in block::make(check(y > 0), check::make(y < 10))` --> not substituted (used twice)